views/js/runner/proxy/cache/proxy.js

```diff
@@ -245,16 +267,46 @@ define([
              *
              * @param {String} action - the action name (ie. move, skip, timeout)
              * @param {Object} actionParams - the parameters sent along the action
+             * @param {Boolean} deferred whether action can be scheduled (put into queue) to be sent in a bunch of actions later.
              * @returns {Promise} resolves with the action result
              */
-            this.requestNetworkThenOffline = function requestNetworkThenOffline(url, action, actionParams){
+            this.requestNetworkThenOffline = function requestNetworkThenOffline(url, action, actionParams, deferred){
                 var testContext = this.getDataHolder().get('testContext');
+                var communicationConfig = self.configStorage.getCommunicationConfig();
 
                 //perform the request, but fallback on offline if the request itself fails
                 var runRequestThenOffline = function runRequestThenOffline(){
-                    return self.request(url, actionParams).then(function(result){
-                        //if the request fails, we should be offline
-                        if(self.isOffline()){
+                    var request;
+                    if (communicationConfig.syncActions && communicationConfig.syncActions.indexOf(action) >= 0) {
+                        request = new Promise(function(resolve) {
+                            self.scheduleAction(action, actionParams).then(function(actionData){
```

Comment:
I think you should explicitely cacth and reject both `self.scheduleAction` and `self.syncOfflineData()` otherwise the current promise won't be canceled in case of internal error or rejection.